### PR TITLE
Create module for uploading resources to SWORD endpoints

### PR DESCRIPTION
**Pull request by linuss**

### CODE REVIEW

File: modules/mod_ginger_sword/mod_ginger_sword.erl [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+format_rsc_uri(RscId, Context) ->
+    sword_url(Context) ++ "/" ++ format_slug(RscId, Context).
+
+render_rdf_body(RscId, Context) ->
+    AC = z_acl:sudo(Context), % We want to publish all properties even if hidden
+    case m_rsc:exists(RscId, Context) of
+        true ->
+            RdfResource = m_rdf:to_triples(RscId, AC),
+            JsonLd = ginger_json_ld:serialize(RdfResource),
+            iolist_to_binary(mochijson2:encode(JsonLd));
```
Comment (ddeboer) [inline modules/mod_ginger_sword/mod_ginger_sword.erl]: Why not use `jsx:encode/1`?

File: modules/mod_ginger_sword/mod_ginger_sword.erl [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+    {Status, _} = Resp,
+    case Status of
+        ok ->
+            Resp;
+        _ ->
+            lager:warning("Http request failed"),
+            lager:info("~p~n", [Request]),
+            lager:info("~p~n", [Resp]),
+            Resp
+    end.
```
Comment (ddeboer) [inline modules/mod_ginger_sword/mod_ginger_sword.erl]: You can use `ginger_http_client` instead for better separation of concerns.

File: modules/mod_ginger_sword/mod_ginger_sword.erl [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+                            Context).
+
+%% @doc Takes a category path and determines if it must be published
+cat_must_publish(IsA) ->
+    ExactCat = case IsA of
+                   [] ->
+                       undefined;
+                   _ ->
+                       lists:last(IsA)
+               end,
```
Comment (ddeboer) [inline modules/mod_ginger_sword/mod_ginger_sword.erl]: I would replace this `case` with a pattern match: `cat_must_publish([])`. See also https://github.com/inaka/erlang_guidelines#avoid-unnecesary-calls-to-length1 (although you’re not using `length/1`, the example is similar).

File: modules/mod_ginger_sword/mod_ginger_sword.erl [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+    lists:any(
+      fun(Cat) ->
+              ExactCat == Cat
+      end,
+      [story]
+     ).
+
+%% @doc Determines if a resource state is published (to the aggregator)
```
Comment (ddeboer) [inline modules/mod_ginger_sword/mod_ginger_sword.erl]: Rather s/is/must be published, right?

File: modules/mod_ginger_sword/mod_ginger_sword.erl (hunk 1 above)
Comment (linuss) [reply modules/mod_ginger_sword/mod_ginger_sword.erl]: Good one, changed!

File: modules/mod_ginger_sword/mod_ginger_sword.erl (hunk 2 above)
Comment (linuss) [reply modules/mod_ginger_sword/mod_ginger_sword.erl]: Done!

File: modules/mod_ginger_sword/mod_ginger_sword.erl (hunk 3 above)
Comment (linuss) [reply modules/mod_ginger_sword/mod_ginger_sword.erl]: Check out the update, it's totally gone now :)

File: modules/mod_ginger_sword/mod_ginger_sword.erl (hunk 4 above)
Comment (linuss) [reply modules/mod_ginger_sword/mod_ginger_sword.erl]: Method removed, no longer applicable :)

**Pull request merged**